### DISCUSSION TITLE
the role 'ansible-role-janus-gateway' was not found

### DISCUSSION THREAD

**Discussion by sujaybane** [tiny-pilot/tinypilot#953] [Q&A] [answered]
Hi,

trying to build tiny pilot version 1.1.0. while running the quick-install script it gives an error as below,

ERROR! the role 'ansible-role-janus-gateway' was not found in /tmp/tmp.zlrudVKHYL/roles:/tmp/tmp.zlrudVKHYL:/tmp/tmp.zlrudVKHYL

The error appears to be in '/tmp/tmp.zlrudVKHYL/mtlynch.tinypilot/tasks/janus.yml': line 4, column 11, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

  import_role:
    name: ansible-role-janus-gateway
          ^ here

and terminates the code. I have checked there is no file created such as janus.yml in the /tmp/ folder. 
can anyone suggest what happens here? what should I have to do at this stage? 

Thanks.

**Comment by mtlynch** [comment, marked as answer]
Thanks for reporting this! I've filed https://github.com/tiny-pilot/tinypilot/issues/954 to track this.

**Reply by sujaybane** [reply]
**Working Background**

Hardware - iMX8M Mini.
tiny-pilot repository - https://github.com/tiny-pilot/tinypilot/tree/1.1.0 (using version 1.1.0).

**performing steps** 

1 - **clone the repository on imx8m mini BSP.**
        git clone https://github.com/tiny-pilot/tinypilot.git -b 1.1.0

2 - **make changes in quick-install scripts.**
      
 **a -** comment out  "sudo apt-get update" because facing an issue related to "gpgv, gpgv2 or gpgv1 required for verification, but neither seems installed" 
      
**b -** change the package "libssl-dev" to "openssl-dev" because it say "package libssl-dev has no installation candidate"

3 -**_run the script_** 
     
 Used the command "./quick-install" to run the script.

4 - **facing issues:**
   
   the role 'ansible-role-janus-gateway' was not found
      
**This is the error log**

      ERROR! the role 'ansible-role-janus-gateway' was not found in 
       /tmp/tmp.zlrudVKHYL/roles:/tmp/tmp.zlrudVKHYL:/tmp/tmp.zlrudVKHYL

       The error appears to be in '/tmp/tmp.zlrudVKHYL/mtlynch.tinypilot/tasks/janus.yml': line 4, column 11, but may
       be elsewhere in the file depending on the exact syntax problem.

       The offending line appears to be:

       import_role:
       name: ansible-role-janus-gateway
       ^ here

5 - **tried solution:**
       
1 - made changes in this line "ansible-galaxy install bitsyai.janus_gateway" in the script
            and try to build.
            this is the reference link: https://stackoverflow.com/questions/39873635/ansible-role-not-found-error
 
 2 - also tried these changes in the script.

           TINYPILOT_ROLE_NAME="ansible-role-tinypilot"
            if [ -d "$TINYPILOT_ROLE_NAME" ]; then
                   pushd "$TINYPILOT_ROLE_NAME"
                  git pull origin master
                  popd
            else
                 TINYPILOT_ROLE_REPO="https://github.com/tiny-pilot/ansible-role-tinypilot.git"
                  git clone "$TINYPILOT_ROLE_REPO" "$TINYPILOT_ROLE_NAME"
            fi

            ansible-galaxy install \
             --force \
            --role-file "${TINYPILOT_ROLE_NAME}/requirements.yml"
            

 but still facing the same error.

can anyone have faced such kind of scenario? Does anyone have an idea how this can be resolved this issue?

Thanks.

**Reply by mtlynch** [reply]
Oh, I thought the 1.1.0 tag was a typo. That release is 18 months old and about 10 releases ago.

TinyPilot is only supported on the Raspberry Pi 4B. Please do not file bugs for installing old versions of TinyPilot or installing TinyPilot on unsupported hardware.

**Reply by sujaybane** [reply]
Hi,

 I have some queries on this.

1 - if Tiny pilot is supported on RPI then can I port it on any other architecture? 
Means can I port it for imx8?

2 - If I try to port the latest version of the tiny pilot on imx8 then is the above-mentioned step are correct?

3 - I have tried with the latest version of the tiny pilot script where In script when they do uname - a or something to check hardware architecture and the script was able to detect the architecture of my platform i.e. aarch64 but still it fails and exit from the script.

Thanks.
